### PR TITLE
Feature/initial python

### PR DESCRIPTION
**Pull request by whaeck**
These are the python bindings for the current state of NDItk.

### CODE REVIEW

File: python/src/MultigroupTable.python.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    "The flux weight record"
+  )
+  .def_property_readonly(
+
+    "reaction_cross_sections",
+    &Table::reactionCrossSections,
+    "The reaction cross section record"
+  );
+
+  // add standard table definitions
```
Comment (joewpeterson) [inline python/src/MultigroupTable.python.cpp:85]: addStandardTableDefinitions doesn't seem to encapsulate everythin in NDItk/src/NDItk/MutigroupTable.hpp::MultigroupTable. The included read.hpp and the print don't seem to be part of this. They would need to be to be a full wrapper, yes?

Comment (whaeck) [reply python/src/MultigroupTable.python.cpp:85]: We only expose fromFile on full data tables, and I haven't added a toFile just yet to write one to a file. The read and print functions will not be exposed directly since these use iterators (and I also believe users shouldn't use these anyway).

File: python/src/read.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  auto iter = string.begin();
+  auto end = string.end();
+
+  std::string key = disco::FreeFormatCharacter::read< std::string >( iter, end );
+  if ( record.keyword() == key ) {
+
+    //! @todo take into account _0 style keys
+    record.read( iter, end, arguments... );
+
+    //! @todo verify the string is now empty
```
Comment (tgsaller) [inline python/src/read.hpp:31]: So read leaves the std::string string empty?

File: src/NDItk/multigroup/FluxWeights/src/read.hpp [hunk 3]
```diff
@@ -6,11 +6,11 @@
  *
  *  @param[in,out] iter     an iterator to the current position in the input
  *  @param[in,out] end      an iterator to the end of the input
- *  @param[in,out] groups   the number of weight values to be read
+ *  @param[in,out] number   the number of weight values
```
Comment (tgsaller) [inline src/NDItk/multigroup/FluxWeights/src/read.hpp:9]: Why the change? The number of weight values is also the number of groups.

File: src/NDItk/multigroup/Metadata.hpp [hunk 4]
```diff
@@ -14,7 +14,7 @@ namespace NDItk {
 namespace multigroup {
 
 /**
- *  @brief Metadata associated to a neutron NDI file
+ *  @brief Metadata associated to multigroup neutron and photon data
```
Comment (tgsaller) [inline src/NDItk/multigroup/Metadata.hpp:17]: associated with*

File: python/src/NDItk.python.cpp [hunk 5]
```diff
@@ -17,5 +23,18 @@ namespace python = pybind11;
  */
 PYBIND11_MODULE( NDItk, module ) {
 
+  python::module::import( "tools" );
+
+  // create the views submodule
+  python::module viewmodule = module.def_submodule(
+
+    "sequence",
```
[the remark this answers is not in the document]
Comment (whaeck) [reply python/src/NDItk.python.cpp:31]: In order to not pollute the python module, I put all the custom views in the sequence module. You can ignore this.

File: python/src/definitions.hpp [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    "Arguments:\n"
+    "    self    the subrecord"
+  );
+}
+
+/**
+ *  @brief Add standard record definitions
+ *
+ *  This adds the following standard properties:
+ *    - keyword, values, size, empty
```
[the remark this answers is not in the document]
Comment (whaeck) [reply python/src/definitions.hpp:69]: I use slightly different documentation strings in both cases, to distinguish between subrecord and record. Hence the code being there twice

File: python/src/MultigroupTable.python.cpp [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  .def(
+
+    python::init< std::string, std::string, std::string, std::string, 
+                  double, double, double, double,
+                  Structure, FluxWeights,
+                  ReactionCrossSections >(),
+    python::arg( "zaid" ), python::arg( "libname" ), python::arg( "source" ),
+    python::arg( "process" ), python::arg( "awr" ), python::arg( "weight" ),
+    python::arg( "temperature" ), python::arg( "dilution" ), 
+    python::arg( "structure" ), python::arg( "flux" ), python::arg( "xs" ),
```
[the remark this answers is not in the document]
Comment (whaeck) [reply python/src/MultigroupTable.python.cpp:43]: I changed the name to reaction_xs and total to total_xs, see fix/review-part3

File: python/src/definitions.hpp [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+  table
+  .def_static(
+
+    "from_file",
+    [] ( const std::string& filename ) -> Table {
+
+      return njoy::NDItk::fromFile< Table >( filename );
+    },
+    "Read an ACE table from a file\n\n"
```
[the remark this answers is not in the document]
Comment (whaeck) [reply python/src/definitions.hpp:140]: Changed in fix/review-part3

File: python/src/multigroup/CrossSection.python.cpp [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+       { return self.crossSections(); },
+    "The cross section values"
+  )
+  .def_property_readonly(
+
+    "number_groups",
+    &Record::numberGroups,
+    "The number of groups defined by this record"
+  );
+
```
[the remark this answers is not in the document]
Comment (whaeck) [reply python/src/multigroup/CrossSection.python.cpp:69]: No, this is a subrecord so I do not have a read function.

File: python/src/multigroup/Metadata.python.cpp [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+    "source_date",
+    &Record::sourceDate,
+    "The source date"
+  )
+  .def_property_readonly(
+
+    "process_date",
+    &Record::processDate,
+    "The process date"
```
[the remark this answers is not in the document]
Comment (whaeck) [reply python/src/multigroup/Metadata.python.cpp:79]: Fixed in fix/review-part3

File: python/src/multigroup/Metadata.python.cpp [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+  )
+  .def_property_readonly(
+
+    "number_reactions",
+    &Record::numberReactions,
+    "The number of reactions defined by this record"
+  )
+  .def(
+
+    "to_string",
```
[the remark this answers is not in the document]
Comment (whaeck) [reply python/src/multigroup/Metadata.python.cpp:119]: records, subrecords and collection of records have to_string. only records should have from_string

File: python/src/read.hpp (hunk 2 above)
Comment (whaeck) [reply python/src/read.hpp:31]: Well, it should.

File: src/NDItk/multigroup/FluxWeights/src/read.hpp (hunk 3 above)
Comment (whaeck) [reply src/NDItk/multigroup/FluxWeights/src/read.hpp:9]: Less changes when i copy/paste that line ;-)

File: src/NDItk/multigroup/Metadata.hpp (hunk 4 above)
Comment (whaeck) [reply src/NDItk/multigroup/Metadata.hpp:17]: That has been changed in fix/review I think, following the sme remark in another PR

**Review by tgsaller — APPROVED** [review]
I am content.
